UnitTests/UnitTests.XamlIslands.UWPApp/XamlIslandsTest_TextToolbar.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.Helpers;
+using Microsoft.Toolkit.Uwp.UI.Controls;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace UnitTests.XamlIslands.UWPApp
+{
+    [STATestClass]
+    public partial class XamlIslandsTest_TextToolbar
+    {
+        private TextToolbar _textToolbar;
+
+        [TestInitialize]
+        public async Task Init()
+        {
+            await App.Dispatcher.ExecuteOnUIThreadAsync(() =>
+            {
+                var richEditBox = new RichEditBox
+                {
+                    PlaceholderText = "Enter Text Here",
+                    TextWrapping = TextWrapping.Wrap,
+                    VerticalContentAlignment = VerticalAlignment.Stretch,
+                    MinHeight = 300,
+                    BorderThickness = new Thickness(1),
+                    SelectionFlyout = null
+                };
+
+                _textToolbar = new TextToolbar
+                {
+                    Editor = richEditBox,
+                    IsEnabled = true,
+                    Format = Microsoft.Toolkit.Uwp.UI.Controls.TextToolbarFormats.Format.RichText
+                };
+
+                var grid = new Grid
+                {
+                    Children =
+                    {
+                        _textToolbar,
+                        richEditBox
+                    },
+                    HorizontalAlignment = HorizontalAlignment.Center,
+                    VerticalAlignment = VerticalAlignment.Center,
+                    Height = 200,
+                    Width = 300
+                };
+                grid.RowDefinitions.Add(new RowDefinition
+                {
+                    Height = GridLength.Auto
+                });
+                grid.RowDefinitions.Add(new RowDefinition());
+                TestsPage.Instance.SetMainTestContent(grid);
+                Grid.SetRow(richEditBox, 1);
+            });
+        }
+
+        [TestMethod]
+        public async Task TextToobar_PopupShowsInCorrectXamlRoot()
+        {
+            await App.Dispatcher.ExecuteOnUIThreadAsync(async () =>
+            {
+                var args = new ShortcutKeyRequestArgs(Windows.System.VirtualKey.K, false, null);
+
+                _textToolbar.GetDefaultButton(Microsoft.Toolkit.Uwp.UI.Controls.TextToolbarButtons.ButtonType.Link).ShortcutRequested(ref args);
```

Comment:
Got a crash here running the tests a 2nd time:
```
Microsoft.Toolkit.Uwp.UI.Controls.TextToolbar.GetDefaultButton(...) returned null.

System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=UnitTests.XamlIslands.UWPApp
  StackTrace:
   at UnitTests.XamlIslands.UWPApp.XamlIslandsTest_TextToolbar.<<TextToobar_PopupShowsInCorrectXamlRoot>b__2_0>d.MoveNext() in 
H:\code\WindowsCommunityToolkit\UnitTests\UnitTests.XamlIslands.UWPApp\XamlIslandsTest_TextToolbar.cs:line 70

```

Reply:
I added a delay to keep this from happening.